src/connections.jl

```diff
@@ -348,9 +363,11 @@ Return a valid PostgreSQL identifier that is unique for the current connection.
 This is mostly used to create names for prepared statements.
 """
 function unique_id(jl_conn::Connection, prefix::AbstractString="")
-    id_number, jl_conn.uid_counter = jl_conn.uid_counter, jl_conn.uid_counter + 1
+    lock(jl_conn.lock) do
+        id_number, jl_conn.uid_counter = jl_conn.uid_counter, jl_conn.uid_counter + 1
```

Comment:
Again not related to this PR,
but the above is just a more confusing way to write:
```
id_number = jl_conn.uid_counter
jl_conn.uid_counter += 1 
```

Reply:
I'll fix that one for this PR